### PR TITLE
:cyclone: Centralizing dev langs config

### LINKED ISSUE #2426: dev_langs metadata
While updating one topic today, I saw that we have jsharp as a dev_lang metadata value and lots of topics on winforms for example are missing other dev_langs. We should update it to remove jsharp and add the appropriate ones. Maybe we can even set those at a global level?

### PR DESCRIPTION
**Pull request by mairaw**
Fixes #2426 

[Internal Review URL](https://review.docs.microsoft.com/en-us/dotnet/standard/?branch=pr-en-us-3235)

### CODE REVIEW

**Comment by mairaw** [comment]
@BillWagner @rpetrusha @JRAlexander 
Ok, this is now working, but I'm wondering if you like the new behavior or not. Let me explain.

For folders that only set one dev_lang per topic such as VB, F# and C#, this works well because we don't have to set that metadata value on each topic anymore and nothing changes on the rendered page. 

However, for folders with multiple values, it means that no matter if the topic has a sample or not, they get a language selector, such as this:
https://review.docs.microsoft.com/en-us/dotnet/standard/net-standard?branch=dev-langs

The alternative would be to set the dev_langs used in each topic for those, similar to what I did in the last commit for when I wanted to override the default value.

Thoughts?

**Review by BillWagner — APPROVED** [review]
I like centralizing the dev_lang attribute.  This will make  it easier to get it correct in all cases.

For your last comment, I'd like to have an engineering change so that dev_langs is optional. For those conceptual topics that are truly language independent, the dev_langs selector would not be rendered. We could control that by not selecting a dev_langs element.  Thoughts?

**Comment by mairaw** [comment]
Yes, @BillWagner, dev_langs is optional. But by centralizing them on the config file, I made them show up in all topics for the framework and standard folders. I could remove the settings for those two folders (or perhaps just choose C# as the default since one language doesn't make the selector appear), so that truly language independent topics wouldn't get that selector. And then we set the language for all topics that are language dependent. Makes sense? I think this would be the best approach IMO (although more work for me LOL). 😆

File: docfx.json [hunk 1]
```diff
@@ -133,6 +133,14 @@
       "langs": {
         "api/**.yml": ["csharp", "vb", "cpp"]
       },
+      "dev_langs": {
+        "docs/core/**/**.md": ["csharp"],
```
Comment (rpetrusha) [inline docfx.json]: Since VB is now supported, shouldn't this be `["csharp", "vb"],`?

**Comment by rpetrusha** [comment]
Possibly I'm missing something, but I don't quite understand the comments about language independence -- shouldn't language independent topics have multiple languages, since they're not language-dependent? Or does language independence here mean something else? 
In terms of the language selector and multiple values, I don't think that it's a significant problem if a page has no code. Offhand, though, I can't find a case in which a page has multiple language selectors but only code in one language. In that case, do you know the behavior, @mairaw?

**Comment by BillWagner** [comment]
@rpetrusha What I meant by 'language independent' was that some of the topics, like the .NET Architecture components, don't have code samples, and don't discuss any language at all. Why show a selector when nothing is going to change by picking a different language?

**Comment by mairaw** [comment]
I talked to @rpetrusha in person and I'm going with plan B, which is to add the values to the pages that have code and leave C# as the default language for standard, core and framework folders. That way, language selector won't show up for "language-independent" topics and we only need to add the dev_lang when we want to override the default behavior.

**Comment by JRAlexander** [comment]
Can we merge this now?

**Comment by mairaw** [comment]
No. It's tagged as WIP because we are going with plan B which requires a lot more work from me to get the experience right.

**Pull request merged**